app/src/main/java/com/mk/tasky/agenda/presentation/detail/event/DetailEventViewModel.kt

```diff
@@ -124,6 +125,15 @@ class DetailEventViewModel @Inject constructor(
                     toTime = event.time
                 )
             }
+            is DetailEventEvents.OnAddPhoto -> {
+                if (state.photos.contains(event.photo)) return
+                val updatedPhotos = mutableListOf<AgendaPhoto>()
+                updatedPhotos.addAll(state.photos)
+                updatedPhotos.add(event.photo)
+                state = state.copy(
+                    photos = updatedPhotos
+                )
+            }
```

Comment:
Just this would do it as well:
```
if(state.photos.contains(event.photo)) return
state = state.copy(
    photos = state.photos + event.photo
)
```

No need for the mutable list :)

Reply:
Oh wow didn't knew that
Nice tip!